src/helper_gitlab_modify.sh

```diff
@@ -294,36 +385,80 @@ delete_gitlab_repository_if_it_exists() {
 }
 
 
-# Structure:gitlab_modify
-#source src/import.sh src/helper_gitlab_modify.sh && delete_existing_repository "sponsor_example" "root"
+#######################################
+# Checks if repository exists in the GitLab server and deletes it. Otherwise an  
+# error is shown.
+# How to run:
+#  source src/import.sh src/helper_gitlab_modify.sh && delete_existing_repository 
+#  "sponsor_example" "root"
+# Local variables:
+#  gitlab_repo_name
+#  gitlab_username
+#  personal_access_token
+# Globals:
+#  GITLAB_PERSONAL_ACCESS_TOKEN
+#  GITLAB_SERVER_HTTP_URL
+# Arguments:
+#  Name of the GitLab repository.
+#  The GitLab username.
+# Returns:
+#  0 if funciton was evaluated succesfull.
+#  183 if an attempt was made to delete a GitLab repository that did not exist.
+# Outputs:
+#  None.
+#######################################
 delete_existing_repository() {
-  repo_name="$1"
-  repo_username="$2"
-  
+  local repo_name="$1"
+  local repo_username="$2"
+
   # load personal_access_token
+  local personal_access_token
   personal_access_token=$(echo "$GITLAB_PERSONAL_ACCESS_TOKEN" | tr -d '\r')
-  
-  #curl -H 'Content-Type: application/json' -H "Private-Token: $personal_access_token" -X DELETE "$GITLAB_SERVER_HTTP_URL"/api/v4/projects/"$repo_username"%2F"$repo_name"
+
+  local output
   output=$(curl -H 'Content-Type: application/json' -H "Private-Token: $personal_access_token" -X DELETE "$GITLAB_SERVER_HTTP_URL"/api/v4/projects/"$repo_username"%2F"$repo_name")
-  
+
   if [  "$(lines_contain_string '{"message":"404 Project Not Found"}' "\${output}")" == "FOUND" ]; then
     echo "ERROR, you tried to delete a GitLab repository that does not exist."
     exit 183
   fi
 }
 
-# Structure:gitlab_modify
-#source src/run_ci_job.sh && clone_repository
+
 # TODO: rename to clone_gitlab_repository_from _local_server
+#######################################
+# Clones the GitLab repository into the GitLab mirror storage location.
+# How to run:
+#  source src/run_ci_job.sh && clone_repository
+# Local variables:
+#  gitlab_repo_name
+#  gitlab_username
+#  gitlab_server_password
+#  gitlab_server
+#  target_directory
+#  output
+# Globals:
+#  None.
+# Arguments:
+#  Name of the GitLab repository.
+#  The GitLab username.
+#  The GitLab server password.
+#  The GitLab server.
+#  The target directory.
+# Returns:
+#  None.
+# Outputs:
+#  None.
+#######################################
 clone_repository() {
-  repo_name=$1
-  gitlab_username=$2
-  gitlab_server_password=$3
-  gitlab_server=$4
-  target_directory=$5
-  
-  # TODO:write test to verify the gitlab username and server don't end with a spacebar character.  
-  
+  local repo_name=$1
+  local gitlab_username=$2
+  local gitlab_server_password=$3
+  local gitlab_server=$4
+  local target_directory=$5
+
+  # TODO:write test to verify the gitlab username and server don't end with a spacebar character.
```

Comment:
This should be included in the TODO's within the properly formatted documentation you created for this function.